### PR TITLE
Add $compile to output html

### PR DESCRIPTION
**Pull request by huei90**

### CODE REVIEW

**Comment by huei90** [comment]
hmmm.. this doesn't pass the test. may need more time to look at this.

**Comment by huei90** [comment]
@Nazanin1369

File: dist/angular-validation.js [hunk 1]
```diff
@@ -387,7 +388,7 @@
       if (element.attr('no-validation-message')) {
         messageElem.css('display', 'none');
       } else if ($validationProvider.showSuccessMessage && messageToShow) {
-        messageElem.html($validationProvider.getSuccessHTML(messageToShow));
+        messageElem.html('').append($compile($validationProvider.getSuccessHTML(messageToShow))(scope));
```
Comment (lvarayut) [inline dist/angular-validation.js]: Should it be `messageElem.html().append($compile($validationProvider.getSuccessHTML(messageToShow))(scope));`?

Comment (huei90) [inline dist/angular-validation.js]: Step1: clear the output html => using `.html('')`
Step2: append a new html element `.append(elm)`

**Comment by lvarayut** [comment]
Alright. How about    `messageElem.html($compile($validationProvider.getSuccessHTML(messageToShow))(scope));` ?

**Comment by huei90** [comment]
@lvarayut I have tried that before
https://github.com/huei90/angular-validation/commit/8eb0a6f40f733ebb3067a9ec53d30a09664a63ee

It will fail because `.html(html_string)` [accepts](http://api.jquery.com/html/#html-htmlString) only `html string`, the content is [node element](http://api.jquery.com/append/#append-content-content) after using `$compile`, so we should use `.append()` in this case.

**Comment by lvarayut** [comment]
It just seems a little bit strange at the first glance. BTW. it' worked perfectly :)

**Comment by huei90** [comment]
Yes, exactly.

**Comment by huei90** [comment]
I have added the test case for the compilation, hope we can trust it :smile:

**Comment by jamhall** [comment]
Ah great, it's been merged. Thanks guys!

**Comment by huei90** [comment]
:+1: @jamhall You can have your repo up to date.

**Pull request merged**